tests/merge_test.py

```diff
@@ -0,0 +1,78 @@
+from kartograf.merge import general_merge
+
+from .generate_data import (
+    build_file_lines,
+    generate_ip_file,
+    generate_file_items,
+    generate_asns,
+    generate_subnets_from_base,
+    make_disjoint
+)
+
+
+def __tmp_paths(tmp_path):
+    return [tmp_path / p for p in ["rpki_final.txt", "irr_final.txt", "out.txt"]]
+
+def test_merge(tmp_path):
+    """
+    Assert that merging two identical files is a no-op.
+    """
+    rpki_data = generate_file_items(100)
+    rpki_path, _, out_path = __tmp_paths(tmp_path)
+    generate_ip_file(rpki_path, rpki_data)
+
+    general_merge(rpki_path, rpki_path, None, out_path)
+
+    with open(out_path, "r") as f:
+        lines = f.readlines()
+        final_ips = [item.split()[0] for item in lines]
+        assert lines == rpki_data
+        assert len(final_ips) == len(rpki_data)
+
+
+def test_merge_disjoint(tmp_path):
+    """
+    Test merging non-overlapping sets of IP networks.
+    """
+    main_data = generate_file_items(100)
+    main_ips = [item.split()[0] for item in main_data]
+    rpki_ips = main_ips[:50]
+    irr_ips = main_ips[50:]
```

Comment:
Just a note (also to myself), since we may remove IRR as a source in the short-mid term when RPKI has adopted broadly enough, we shouldn't hardcode it in the tests unless necessary. Here I think the variables could have been named `base` and `extra` as well and that's a bit more future-proof.